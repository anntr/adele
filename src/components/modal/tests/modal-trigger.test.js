import React from 'react';
import ModalTrigger from '../modal-trigger';

const MockChild = () => <div />;
const MockContent = () => <div />;
const openModalMock = jest.fn();
const mockContext = { context: { openModal: openModalMock } };

it('renders trigger correctly', () => {
  const wrapper = mount(<ModalTrigger>{() => <MockChild />}</ModalTrigger>, mockContext);

  expect(wrapper).toMatchSnapshot();
});

it('passes content via openModal handler to children', () => {
  const wrapper = mount(
    <ModalTrigger content={() => <MockContent />}>
      {openModal => <MockChild openModal={openModal} />}
    </ModalTrigger>,
    mockContext,
  );

  wrapper.find(MockChild).prop('openModal')();

  expect(openModalMock).toBeCalledWith(wrapper.prop('content'));
});
