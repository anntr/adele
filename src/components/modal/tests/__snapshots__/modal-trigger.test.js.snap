// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders trigger correctly 1`] = `
<getContext(Component)>
  <Component
    openModal={[Function]}
  >
    <MockChild>
      <div />
    </MockChild>
  </Component>
</getContext(Component)>
`;
